core/src/main/java/org/jdbi/v3/core/statement/TemplateEngine.java

```diff
@@ -17,6 +17,8 @@
  * Renders an SQL statement from a template.
  *
  * @see DefinedAttributeTemplateEngine
+ *
+ * Note for implementors: define an appropriate constructor or constant for SqlObject's UseTemplateEngine annotation!
```

Comment:
We should actually define what an appropriate constructor or constant looks like.

Reply:
Is expecting implementors to give the UseTemplateEngine source code a quick look too much to expect? I'm worried if we document the specifics in this separate class, someone will modify UseTemplateEngine's behavior and forget to keep this documentation in sync